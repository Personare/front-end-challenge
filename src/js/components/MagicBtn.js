import React, { Component } from 'react';

class MagicBtn extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <button className='MagicBtn is-pink'>
                <svg>
                    <defs>
                        <linearGradient id="border-purple" x1="0" y1="0" x2="0" y2="1" spreadMethod="pad">
                            <stop offset="0%" stopColor="#fce9ff"></stop>
                            <stop offset="100%" stopColor="#7d629f"></stop>
                        </linearGradient>

                        <linearGradient id="light" x1="0.000" y1="0.000" x2="0.968" y2="1.030" spreadMethod="pad">
                            <stop offset="90%" stopColor="#fbeef5" stopOpacity="0"></stop>
                            <stop offset="100%" stopColor="#fbeef5" stopOpacity="1"></stop>
                        </linearGradient>

                        <filter id="glow-purple" x="-40%" y="-40%" width="180%" height="180%">
                            <feColorMatrix type="matrix" values="0.91 0 0 0   0
                                       0 0.44 0 0   0
                                       0 0 1 0   0
                                       0 0 0 1 0"></feColorMatrix>
                            <feGaussianBlur stdDeviation="3" result="coloredBlur"></feGaussianBlur>
                            <feMerge>
                                <feMergeNode in="coloredBlur"></feMergeNode>
                                <feMergeNode in="coloredBlur"></feMergeNode>
                                <feMergeNode in="SourceGraphic"></feMergeNode>
                            </feMerge>
                        </filter>
                    </defs>

                    <g filter="url(#glow-purple)">
                        <rect className="border" width="100%" height="100%" rx="10" ry="10" fill="transparent" stroke="url(#border-purple)" strokeWidth="1"></rect>
                        <rect className="light" width="100%" height="100%" rx="10" ry="10" fill="transparent" stroke="url(#light)" strokeWidth="2" opacity="0.6"></rect>
                    </g>
                </svg>

                {this.props.text}
            </button>
        );
    }
}

export default MagicBtn;
