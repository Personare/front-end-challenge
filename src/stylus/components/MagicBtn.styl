.MagicBtn
    cursor pointer
    display inline-block
    color #fedeff
    padding-top: 13px;
    padding-bottom: 13px;
    width 100%
    max-width 150px
    font-size 2em
    position relative
    transition background .6s cubic-bezier(.25,.46,.45,.94)

    &:hover
        background #17083a

    &:focus
        outline none

    svg
        position absolute
        top 2px
        left 0px
        padding 1px
        height 100%
        width 100%
        overflow visible
