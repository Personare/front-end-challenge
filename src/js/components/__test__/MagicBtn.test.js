import React from 'react';
import { shallow } from "enzyme";
import MagicBtn from '../MagicBtn';

describe('MagicBtn', () => {

    test('should render correctly width', () => {
        const wrapper = shallow(<MagicBtn text='iniciar' />);
        expect(wrapper).toMatchSnapshot();
    });

})
