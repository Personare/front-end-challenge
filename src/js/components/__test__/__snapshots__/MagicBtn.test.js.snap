// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicBtn should render correctly width 1`] = `
<button
  className="MagicBtn is-pink"
>
  <svg>
    <defs>
      <linearGradient
        id="border-purple"
        spreadMethod="pad"
        x1="0"
        x2="0"
        y1="0"
        y2="1"
      >
        <stop
          offset="0%"
          stopColor="#fce9ff"
        />
        <stop
          offset="100%"
          stopColor="#7d629f"
        />
      </linearGradient>
      <linearGradient
        id="light"
        spreadMethod="pad"
        x1="0.000"
        x2="0.968"
        y1="0.000"
        y2="1.030"
      >
        <stop
          offset="90%"
          stopColor="#fbeef5"
          stopOpacity="0"
        />
        <stop
          offset="100%"
          stopColor="#fbeef5"
          stopOpacity="1"
        />
      </linearGradient>
      <filter
        height="180%"
        id="glow-purple"
        width="180%"
        x="-40%"
        y="-40%"
      >
        <feColorMatrix
          type="matrix"
          values="0.91 0 0 0   0
 0 0.44 0 0   0
 0 0 1 0   0
 0 0 0 1 0"
        />
        <feGaussianBlur
          result="coloredBlur"
          stdDeviation="3"
        />
        <feMerge>
          <feMergeNode
            in="coloredBlur"
          />
          <feMergeNode
            in="coloredBlur"
          />
          <feMergeNode
            in="SourceGraphic"
          />
        </feMerge>
      </filter>
    </defs>
    <g
      filter="url(#glow-purple)"
    >
      <rect
        className="border"
        fill="transparent"
        height="100%"
        rx="10"
        ry="10"
        stroke="url(#border-purple)"
        strokeWidth="1"
        width="100%"
      />
      <rect
        className="light"
        fill="transparent"
        height="100%"
        opacity="0.6"
        rx="10"
        ry="10"
        stroke="url(#light)"
        strokeWidth="2"
        width="100%"
      />
    </g>
  </svg>
  iniciar
</button>
`;
